### PR TITLE
feat(collections): time-driven bells + host-driven recurrence

### PR DESCRIPTION
**Pull request by snakajima**
## Summary

Adds two composable, schema-declared primitives to the **collections** app platform, both built on the existing convergent completion-notification reconciler. Together they let a collection express a **recurring obligation** (rent, subscriptions, renewals) with no cadence engine and no per-feature host code — the philosophy the collections line is built on.

This was prototyped live during development: the agent authored a multi-payee `recurring-bills` collection straight from the help doc, the bell fired on the lead date, and marking a bill paid auto-created next month's — each payee an independent monthly chain in one collection.

### 1. `triggerField` — time-gated bell
A `date` field that **delays** the completion bell until that date arrives, instead of firing on create. Optional `triggerLeadDays` fires it N days early ("remind me 10 days before it's due").

- Firing is **derived from the clock, not stored** — a new 1-minute wall-clock tick (modeled on the existing rediscovery timer in `watcher.ts`) re-reconciles time-dependent collections. Missed-fire-while-the-server-was-down is handled for free; pushing the date out retracts a premature bell.
- Day granularity, server-local timezone. Requires the completion pair.

### 2. `spawn` — host-driven recurrence
When a record matches `spawn.when` (default: it's "done"), the host creates the **next** record with `triggerField` advanced by `every`.

- **Convergent, not event-driven**: reconciles on a *predicate* ("matches AND successor absent"); the successor's deterministic id + contents are a pure function of `(source, rule)`, and creation is **create-if-absent** — so re-observing the predicate writes the successor exactly once. No stored "fired" flag, no double-spawn.
- **Drift-free month math** in a new pure `spawn.ts`: `every` supports `day`/`week`/`month`/`year` + `interval`, with a rule-anchored `dayOfMonth` (1–31 or `"last"`) so "31st of every month" → 31 → 28/29 → 31 → 30 … correctly. `triggerLeadDays` composes for free (lead applied at fire time, so every cycle fires the same days early).
- **Forward-only**: un-doing the source doesn't retract an already-created successor; to stop a recurrence, move the source to a status outside `spawn.when`.

## Scope boundary
Covers the **periodic** class of obligation. Escalating, multi-stage reminders over a long prep window (info → warning → urgent) are a **deliberate non-goal** — that stays with Encore-as-bespoke, because that escalation-state complexity is exactly what would break the pure-convergence property this design relies on. See `plans/feat-collections-time-trigger.md`.

## Changes
- **Source**: `types.ts`, `discovery.ts` (Zod + refines), `notifications.ts` (`now` param + gate), `watcher.ts` (wall-clock tick), `spawn.ts` (new, pure).
- **Tests** (+12, 5876 total, 0 fail): `test_spawn.ts` (new — civil-date math incl. day-31 no-drift, leap years, lead-day boundaries, deterministic id, create-if-absent idempotency), plus trigger/lead/spawn cases in `test_notifications.ts` and validation cases in `test_discovery.ts`.
- **Docs**: `server/workspace/helps/collection-skills.md` (agent-facing schema DSL reference), `plans/feat-collections-time-trigger.md` (design).
- **UX**: a recurring-bills sample query on the Personal role.

## Verification
`yarn format`, `yarn lint`, `yarn typecheck`, `yarn build`, `yarn test` — all clean (5876 pass / 0 fail).

## Known follow-ups (out of scope, noted in the plan)
- The standalone `/collections` table is REST-pull, so a spawned record needs a UI refresh to appear (the bell updates live; the list doesn't). Giving the list a change-notification channel like the bell's is a separate feature.
- `every` and `triggerLeadDays` are schema-wide, so one collection can't mix *cadences* (monthly + annual). Per-record cadence would be a future extension.
- Time-of-day (`datetime`) and `spawn.until` are deferred.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **New Features**
  * Time-gated completion notifications (trigger date + optional lead days)
  * Automatic recurring record generation with configurable schedules and deterministic successor IDs
  * Field carry-over and automatic value setting for generated records
  * Periodic background tick to re-evaluate time-driven collections

* **Bug Fixes**
  * Suppress/clear notifications when triggers aren’t due; gracefully handle unparseable trigger values

* **Tests**
  * Comprehensive unit and integration tests for validation, triggering, spawning, and edge cases

* **Documentation**
  * Updated collection configuration guide and checklist for time-trigger and recurrence features

### CODE REVIEW

File: src/config/roles.ts [hunk 1]
```diff
@@ -115,6 +115,7 @@ export const ROLES: Role[] = [
       "Create a contacts collection with name, company, title, email, phone, notes, and a business-card image. When I attach a photo of a business card, read the details off it and add a new contact.",
       "Create a reading-list collection with a title, a URL field, and a Read checkbox. While Read is unchecked, keep each item in the bell notifications, labeled with its title.",
       "Create a restaurants collection with name, cuisine, neighborhood, a website URL, a phone number, a Visited checkbox, a 1-to-5 rating, and notes. Hide the rating until I've marked a place as visited — there's nothing to rate before I've been.",
+      "Create a bills collection to track recurring payments — payee, amount, due date, and status. Remind me 10 days before each bill is due, and when I mark one paid, automatically set up next month's bill.",
```
[the remark this answers is not in the document]
Comment (snakajima) [reply src/config/roles.ts:118]: This is by design. We have a mechanism to auto translate those sample queries.

**Comment by snakajima** [comment]
Pushed `9f338a0a` addressing the review:

**🟠 `discovery.ts` — runaway respawn (valid, fixed).** Good catch. A successor born already matching its own spawn predicate would re-spawn on its first reconcile → unbounded chain. Guarded at two layers:
- **Static** (`spawnSuccessorStartsInert` Zod refine) — generalized beyond the default-`when` case to also cover an explicit `when` and the `carry` path: rejects when `set[predicateField]` lands on a matching value, or when the predicate field is `carry`'d (the successor would inherit the matching source value).
- **Runtime** (`maybeSpawnSuccessor`) — belt-and-suspenders: skip + warn if the computed successor matches `when`.
- Tests added for both (discovery rejects; runtime refuses).

**🟡 `collection-skills.md` — timezone wording.** Resolved the *other* direction. There is no separate "workspace timezone" source — the code uses the process/server-local clock (`now.getFullYear()/getMonth()/getDate()`), same as Encore. So the help doc's "server's local timezone" is the accurate term; the **plan** had the stale "workspace timezone" wording, which I've corrected so plan + help + code now agree.

**🟡 plan MD040 — unlabeled fences.** Added `text` language tags to the four flagged fences; all opening fences now carry a language.

**`roles.ts` i18n — skipped (by design).** The `queries` arrays are auto-translated by a dedicated pipeline, as confirmed above.

`yarn format`/`lint`/`typecheck`/`build`/`test` all clean — 5879 pass / 0 fail.
